code/modules/crafting/working/butter_churn.dm

```diff
@@ -0,0 +1,43 @@
+/obj/structure/working/butter_churn
+	name         = "butter churn"
+	desc         = "A simple contraption for churning cream into butter."
+	icon         = 'icons/obj/structures/butter_churn.dmi'
+	product_type = /obj/item/food/dairy/butter/stick
+	work_sound   = /datum/composite_sound/loom_working
+	atom_flags   = ATOM_FLAG_OPEN_CONTAINER
+
+/obj/structure/working/butter_churn/Initialize()
+	. = ..()
+	initialize_reagents()
+
+/obj/structure/working/butter_churn/initialize_reagents(populate)
+	create_reagents(200)
+	. = ..()
+
+/obj/structure/working/butter_churn/try_start_working(mob/user)
+
+	if(!reagents?.has_reagent(/decl/material/liquid/drink/milk/cream, 20))
+		to_chat(user, SPAN_NOTICE("\The [src] does not have enough cream to churn into butter."))
+		return FALSE
+
+	start_working()
+	var/decl/chemical_reaction/butter_recipe = GET_DECL(/decl/chemical_reaction/recipe/food/dairy/butter)
+	var/processed = 0
+	while(user.do_skilled(1.5 SECONDS, work_skill, src) && reagents?.has_reagent(/decl/material/liquid/drink/milk/cream, 20))
```

Comment:
was going to ask why this didn't use `butter_recipe.can_happen()`, but it seems that this is specifically to avoid requiring enzyme.

personally i'd probably make an enzymeless subtype of the butter recipe that overrides `can_happen` to like... check if we're in a butter churn that is currently working? but also that'd be more complicated and it's whatever tbh